suite-native/graph/src/components/AxisLabel.tsx

```diff
@@ -10,34 +10,53 @@ type AxisLabelProps = {
     value: number;
 };
 
+type AxisLabelStyleProps = {
+    x: number;
+    isOverflowing: boolean;
+};
+
+const SCREEN_WIDTH = Dimensions.get('screen').width;
+
 export const MAX_CLAMP_VALUE = 90;
 
-const axisLabelStyle = prepareNativeStyle<Pick<AxisLabelProps, 'x'>>((_, { x }) => ({
+const axisLabelStyle = prepareNativeStyle<AxisLabelStyleProps>((_, { x, isOverflowing }) => ({
     position: 'absolute',
     left: `${x}%`,
+    extend: {
+        // Position the axis label from the right if it's overflowing the screen width.
+        condition: isOverflowing,
+        style: {
+            left: undefined,
+            right: 0,
```

Comment:
Should be handled for left side as well. You could just use `right: 0` if it's max value and `left: 0` if it's min value

Reply:
why for the left side? The `x` prop will be never lower than 0.